components/Banner/examples/supplemental/MDCBannerView.m

```diff
@@ -14,29 +14,42 @@
 
 #import "MDCBannerView.h"
 
+#import "MDCBannerViewConstants.h"
 #import "MDCBannerViewLayout.h"
 #import "MaterialButtons.h"
+#import "MaterialTypography.h"
 
-static const CGFloat kIconImageContainerSideLength = 40.0f;
-static const CGFloat kIconImageSideLength = 24.0f;
-static const CGFloat kTextColorOpacity = 0.87f;
-static const CGFloat kTextFontSize = 14.0f;
+static const CGFloat kIconImageViewSideLength = 40;
 static const NSInteger kTextNumberOfLineLimit = 3;
+#if DEBUG
 static const NSUInteger kNumberOfButtonsLimit = 2;
+#endif
+
+static const uint32_t MDCBannerDefaultBackgroundColor = 0xFFFFFFFF;
+
+static inline UIColor *MDCColorFromRGB(uint32_t rgbValue) {
+  return [UIColor colorWithRed:((CGFloat)((rgbValue & 0xFF0000) >> 16)) / 255
+                         green:((CGFloat)((rgbValue & 0x00FF00) >> 8)) / 255
+                          blue:((CGFloat)((rgbValue & 0x0000FF) >> 0)) / 255
+                         alpha:1];
+}
 
 @interface MDCBannerView ()
 
 @property(nonatomic, readwrite, weak) UILabel *textLabel;
-@property(nonatomic, readwrite, weak) UIView *iconImageViewContainer;
 @property(nonatomic, readwrite, weak) UIImageView *iconImageView;
-@property(nonatomic, readwrite, copy) NSArray<MDCButton *> *buttons;
+@property(nonatomic, readwrite, strong) UIView *containerView;
 
 @property(nonatomic, readwrite, strong) MDCBannerViewLayout *layout;
+@property(nonatomic, readwrite, assign) MDCBannerViewLayout *cachedLayout;
+@property(nonatomic, readwrite, assign) BOOL isLayoutDirty;
```

Comment:
Tip: Remove this an add `setNeedsLayout` where appropriate.

Reply:
Same reason as mentioned above in https://github.com/material-components/material-components-ios/pull/6830#discussion_r264361551. We can talk offline on it.